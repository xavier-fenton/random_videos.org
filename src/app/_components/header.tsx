import React from 'react';
import { useFireBaseAuth } from '../_providers/FireBaseAuthProvider';
import { useRouter } from 'next/navigation';
import SignInButton from './_subcomponents/signInButton';
import SignUpButton from './_subcomponents/signUpButton';

export default function Header(){

  const { user } = useFireBaseAuth()
  
  const router = useRouter()
  const handleClick = () => {
    router.push(`/profile/${user?.email}`)
  }
 
  return (
    <div className='flex h-[100px] justify-between border border-black'>
     {user ? <span></span> : <SignUpButton />} 
      <div className='flex justify-center items-center'>welcome to random_videos.org</div>
      { user ? <div className='flex justify-center items-center pr-4'><button onClick={handleClick}>{user.email}</button></div> : <div className='flex justify-center items-center pl-4'><SignInButton/></div>}

    </div>
  );
};
