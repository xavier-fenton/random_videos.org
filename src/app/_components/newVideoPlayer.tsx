import React, { useState } from 'react'

const NewVideoPlayer = ({ source, visibility }: any) => {
  const [buttonColour, setButtonColour] = useState<string>('bg-green-400')
  const [textColour, setTextColour] = useState<string>('text-green-400')
  const [adjustmentButtonColour, setAdjustmentButtonColour] =
    useState<string>('bg-slate-400')
  const [adjustmentTextColour, setAdjustmentTextColour] =
    useState<string>('text-slate-400')
  const [adjustmentSize, setAdjustmentSize] = useState<string>('px-20')
  const [adjusted, setAdjusted] = useState<boolean>(false)

  const handleClose = () => {
    visibility(false)
  }
  const handleAdjustment = () => {
    if (adjusted) {
      setAdjustmentSize('px-20')
    } else {
      setAdjustmentSize('px-0')
    }
    setAdjusted(!adjusted)
  }

  return (
    <div
      className={`absolute z-10 drop-shadow-2xl top-0 h-[100dvh] ${
        adjusted ? adjustmentSize : 'px-20'
      } transition ease-in-out duration-300  `}
    >
      <div className="bg-gray-200 rounded-lg gap-2 py-2 h-[100%]">
        <div className="flex justify-end gap-2 p-2">
          <button
            className={`${adjustmentButtonColour} ${adjustmentTextColour} rounded-full text-xs px-2 `}
            onClick={handleAdjustment}
            onMouseEnter={() => {
              setAdjustmentButtonColour('bg-slate-300')
              setAdjustmentTextColour('text-slate-300')
            }}
            onMouseLeave={() => {
              setAdjustmentButtonColour('bg-slate-400')
              setAdjustmentTextColour('text-slate-400')
            }}
          >
            x
          </button>
          <button
            className={`${buttonColour} ${textColour} rounded-full text-xs px-2 `}
            onClick={handleClose}
            onMouseEnter={() => {
              setButtonColour('bg-red-600')
              setTextColour('text-red-600')
            }}
            onMouseLeave={() => {
              setButtonColour('bg-green-400')
              setTextColour('text-green-400')
            }}
          >
            x
          </button>
        </div>
        <div className="flex flex-col items-center bg-gray-200 rounded-b-lg">
          <video
            className="h-auto w-[80%] rounded-md shadow-lg"
            controls
            src={source}
          />
          <div>
            <div className="pt-2 px-4">title</div>
            <div className="px-4 pb-4">description</div>
            <div className="px-[20px] pb-[20px]">
              <div className="px-4 w-[75%] rounded-lg border border-black">
                Eu deserunt amet adipisicing occaecat est consequat cillum
                labore in do. Nisi eu ea culpa amet magna non ullamco magna.
                Laboris esse magna culpa sint labore et in eu eu pariatur.
                Ullamco eu cillum pariatur dolor laborum ipsum nulla. Dolor
                laboris amet qui sint quis aliqua ea nostrud ex nulla consequat.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default NewVideoPlayer
