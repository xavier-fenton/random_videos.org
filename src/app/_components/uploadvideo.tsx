import React from 'react'

const Uploadvideo = () => {
  return (
    <div className="h-[100dvh] w-full p-4 border border-x-0">
      <div className="w-fit flex flex-col gap-4">
        <label>Upload a video: </label>
        <input className="text-sm" type="file" accept="video/*" />
      </div>
    </div>
  )
}

export default Uploadvideo
