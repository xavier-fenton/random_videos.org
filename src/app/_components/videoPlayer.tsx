import React, { use, useEffect, useState } from 'react'
import { useVideoContext } from '../../lib/_context/videoContext'

export function VideoPlayer() {
  const { currentVideo, loader2 } = useVideoContext()
  const [video, setVideo] = useState<any>()

  useEffect(() => {
    if (currentVideo.length === 0) {
      setVideo(false)
    } else currentVideo.length === 1 && setVideo(currentVideo)
  }, [currentVideo, setVideo])

  return (
    <>
      {!loader2 && video ? (
        video.map(
          (video_info: {
            id: number
            title: string
            testkey: number
            video_file: string
          }) => {
            return (
              <div
                key={video_info.testkey}
                className="flex flex-col gap-4 w-full px-4 "
              >
                <div className="text-center pt-4 underline">
                  {video_info.title}
                </div>
                <div className="flex items-center justify-center w-full h-fit">
                  <video
                    id="video_test"
                    autoPlay
                    controls
                    src={video_info.video_file}
                  />
                </div>
                <div className="text-center underline">{video_info.title}</div>
              </div>
            )
          }
        )
      ) : (
        <div className='flex justify-center items-center w-[100%]'>
          <div className="loader">Loading...</div>
        </div>
      )}
    </>
  )
}
