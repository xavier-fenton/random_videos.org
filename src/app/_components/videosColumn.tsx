import React, { useState } from 'react'
import { user } from '../testUsers'
import { useVideoContext } from '../_context/videoContext'
import { useParams } from 'next/navigation'

export interface Props {
  usersVideoCollection: user[]
  userArchive: userArchive
}

type userArchive = {
  username: string
  userVideos: []
}

export const VideoColumn: React.FC<Props> = ({
  usersVideoCollection,
  userArchive,
}) => {
  const params = useParams<{ username: string }>()

  const { setVideoPlayer, loader2, setLoader2 } = useVideoContext()
  

  const userVideosJSX = usersVideoCollection?.map((userVideos: any) =>
    userVideos.map((video: any) => {
      return (
        <div
          key={video.testkey}
          className="flex border-2 border-[#d8cbcb] w-full h-full bg-[#d8cbcb] text-center justify-center items-center"
        >
          <button
            onClick={() => {
              setLoader2(true)
              setTimeout(() => {
                setVideoPlayer(video)
                setLoader2(false)
              }, 1500)
            }}
          >
            <video key={video.testkey} className="h-full">
              <source src={video.video_file} type="video/mp4" />
            </video>
          </button>
        </div>
      )
    })
  )

  

  return (
    <>
      {userArchive?.username && userArchive?.username === params?.username ? (
        <div className="flex flex-col gap-5 border border-x-0 w-full h-[100dvh] p-5">
          <div className="text-xs bg-[#dedede] w-fit p-[2px] px-4 rounded-2xl text-[#424040]">all your random_videos that are out there:</div>
          <div className="grid grid-cols-1 md:grid-cols-3 lg:grid-cols-4 gap-4">

            {userArchive.userVideos ? (
              userArchive.userVideos.map((video_info: any) => {
                return (
                  <video key={video_info.testkey} className="h-full w-full" src={video_info.video_file} ></video>
                )
              })
            ) : (
              <div className="loader">Loading</div>
            )}
          </div>
        </div>
      ) : (
        <div className="flex flex-col gap-3 border w-full md:w-[50%] md:border-r-black h-[100dvh] p-5">
          <div className="text-center">random_videos</div>
          <div className="grid grid-cols-2 gap-4">
            {userVideosJSX ? (
              userVideosJSX
            ) : (
              <div className="loader">Loading</div>
            )}
          </div>
        </div>
      )}
    </>
  )
}
