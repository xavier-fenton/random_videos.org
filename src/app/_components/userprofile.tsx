import React, { useState } from 'react'
import { VideoPlayer } from './videoPlayer'
import Uploadvideo from './_subcomponents/userProfileComponents/uploadvideo'
import { useFireBaseAuth } from '../../lib/_providers/FireBaseAuthProvider'
import UserProfileHeader from './_subcomponents/userProfileComponents/userProfileHeader'
import { useRouter } from 'next/router'
import UserVideoColumn from './_subcomponents/userProfileComponents/userVideoColumn'

export const UserProfile = () => {
  const [state, setState] = useState('user archive:')

  const router = useRouter()
  const { username } = router.query
  const { user } = useFireBaseAuth()

  const handleButtonClicked = (e: Event | any) => {
    setState(e.target.innerHTML)
  }
  const handleBackButton = (e: Event | any) => {
    e.preventDefault()
    router.push('/')
  }

  return (
    <>
      {user?.email === username ? (
        <div className="pt-4 flex flex-col">
          <div className="flex">
            <div className="flex flex-row gap-4">
              <button
                className="text-[#eee] text-sm bg-gray-400 rounded-r-2xl px-2"
                onClick={handleBackButton}
              >
                back to explorer
              </button>
              <UserProfileHeader user={user} />
            </div>
            <div
              onClick={handleButtonClicked}
              id="user_archive"
              className="flex justify-center items-center cursor-pointer w-fit  bg-[#dedede] rounded-t-lg px-2 text-xs py-[2px]"
            >
              user archive:
            </div>
            <div
              onClick={handleButtonClicked}
              id="user_upload"
              className="flex justify-center items-center cursor-pointer w-fit bg-[#f1f1f1] rounded-t-lg px-2 text-xs py-[2px]"
            >
              user upload
            </div>
          </div>
          <div className="flex flex-col w-[100%] h-[100dvh]">
            {state === 'user archive:' ? (
              <UserVideoColumn />
            ) : state === 'user upload' ? (
              <Uploadvideo />
            ) : (
              <VideoPlayer />
            )}
          </div>
        </div>
      ) : (
        <div>Wrong place go back</div>
      )}
    </>
  )
}
