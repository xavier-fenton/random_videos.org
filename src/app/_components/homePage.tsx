import React, { useEffect } from 'react'
import { useVideoContext } from '../../lib/_context/videoContext'
import { VideoColumn } from './videosColumn'
import NotUser from './notUser'

const HomePage = () => {
  const { loader, setLoader, userVideos } = useVideoContext()
  
  useEffect(() => {
    // Update loader state after userVideos is loaded
    if (userVideos === undefined) {
      setLoader(true)
    } else {
      setLoader(false)
    }
  }, [userVideos, setLoader])

  return (
    <>
      {loader ? (
        <div className='flex justify-center items-center pt-10 h-[100dvh]'>
          <div className="loader">Loading...</div>
        </div>
      ) : (
        <>
          {!loader && userVideos.length ? (
            <>
                <VideoColumn usersVideoCollection={userVideos} />
            </>
          ) : (
            <NotUser />
          )}
        </>
      )}
    </>
  )
}

export default HomePage
