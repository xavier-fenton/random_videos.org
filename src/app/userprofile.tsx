import React, { useState } from 'react'
import { useVideoContext } from './_context/videoContext'
import '../app/globals.css'
import { VideoPlayer } from './videoPlayer'
import Uploadvideo from './uploadvideo'
import { TestUsers as userVideos } from './testUsers'
import { VideoColumn } from './videosColumn'

type Props = {
  user: any
}

export const UserProfile: React.FC<Props> = ({ user }) => {
  const [state, setState] = useState('user archive:')
  const [userArchive, setUserArchive] = useState(user)
  
  

  const handleButtonClicked = (e: any) => {
    setState(e.target.innerHTML)
  }


  return (
    <div className="pt-4 flex flex-col">

      <div className="flex pl-4">
      <div className="flex text-sm pr-4 items-center"> <div className='text-xs'>hello</div> <div className='text-2xl'>{user.username}</div></div>
        <div onClick={handleButtonClicked} id="user_archive" className="flex justify-center items-center cursor-pointer w-fit  bg-[#dedede] rounded-t-lg px-2 text-xs py-[2px]">
          user archive:
        </div>
        <div onClick={handleButtonClicked} id="user_upload" className="flex justify-center items-center cursor-pointer w-fit bg-[#f1f1f1] rounded-t-lg px-2 text-xs py-[2px]">
          user upload
        </div>
      </div>
      <div className="flex flex-col w-[100%] h-[100dvh]">
          {state === 'user archive:' ? <VideoColumn userArchive={userArchive}/> : state === 'user upload' ? <Uploadvideo /> : <VideoPlayer userArchive={userArchive}/> }
         
      </div>
    </div>
  )
}
