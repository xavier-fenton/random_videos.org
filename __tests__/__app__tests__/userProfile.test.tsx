import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import { UserProfile } from '@component/app/_components/userprofile'
import mockRouter from 'next-router-mock';
import { useRouter } from 'next/router';
import { AppRouterContextProviderMock } from '../../app-router-context-provider-mock';


jest.mock('next/router', () => jest.requireActual('next-router-mock'))

const ExampleComponent = ({ href = '' }) => {
  const router = useRouter();
  return (
    <button onClick={() => router.push(href)}>
      The current route is: "{router.asPath}"
    </button>
  );
}

describe('next-router-mock', () => {
  it('mocks the useRouter hook', () => {
    // Set the initial url:
    mockRouter.push("/initial-path");

    // Render the component:
    render(<ExampleComponent href="/foo?bar=baz" />);
    expect(screen.getByRole('button')).toHaveTextContent(
      'The current route is: "/initial-path"'
    );

    // Click the button:
    fireEvent.click(screen.getByRole('button'));

    // Ensure the router was updated:
    expect(mockRouter).toMatchObject({
      asPath: "/foo?bar=baz",
      pathname: "/foo",
      query: { bar: "baz" },
    });
  });
});

describe('UserProfile snapshot', () => {
  it('matches snapshot', () => {
    const push = jest.fn()
    const { container } = render(<AppRouterContextProviderMock router={{ push }}><UserProfile /></AppRouterContextProviderMock>)

    expect(container).toMatchSnapshot()
  })
})