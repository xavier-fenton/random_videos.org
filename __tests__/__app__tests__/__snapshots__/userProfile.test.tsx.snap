// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile snapshot matches snapshot 1`] = `
<div>
  <div
    class="pt-4 flex flex-col"
  >
    <div
      class="flex"
    >
      <div
        class="flex flex-row gap-4"
      >
        <button
          class="text-[#eee] text-sm bg-gray-400 rounded-r-2xl px-2"
        >
          back to explorer
        </button>
        <div
          class="flex text-sm px-4 items-center gap-4"
        >
          <div
            class="text-xs"
          >
            hello
          </div>
          <div
            class="text-2xl"
          />
          <button>
            sign out
          </button>
        </div>
      </div>
      <div
        class="flex justify-center items-center cursor-pointer w-fit  bg-[#dedede] rounded-t-lg px-2 text-xs py-[2px]"
        id="user_archive"
      >
        user archive:
      </div>
      <div
        class="flex justify-center items-center cursor-pointer w-fit bg-[#f1f1f1] rounded-t-lg px-2 text-xs py-[2px]"
        id="user_upload"
      >
        user upload
      </div>
    </div>
    <div
      class="flex flex-col w-[100%] h-[100dvh]"
    >
      <div
        class="flex flex-col gap-5 border border-x-0 w-full h-[100dvh] p-5 items-center"
      >
        <div
          class="text-xs bg-[#dedede] w-fit p-[2px] px-4 rounded-2xl text-[#424040]"
        >
          all your random_videos that are out there:
        </div>
        <div
          class="loader"
        >
          loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;
